src/formatters/diffs/string.js

```diff
@@ -0,0 +1,143 @@
+import Consts from './const';
+
+const DIFF_MARKER_SYMBOL  = '^';
+const EMPTY_STRING_LENGTH = '""'.length;
+
+var cutNewLines = function (code) {
+    return typeof code === 'string' ? code.replace(/(\r\n|\n|\r)/gm, '\\n') : code;
+};
+
+var getMarkerStr = function (offset) {
+    return [(new Array(offset + 1)).join(' '),
+        (new Array(2)).join(DIFF_MARKER_SYMBOL)].join('');
+};
+
+var OVERFLOW_TYPES = {
+    NO_OVERFLOW:         'NO_OVERFLOW',
+    LEFT_OVERFLOW:       'LEFT_OVERFLOW',
+    RIGHT_OVERFLOW:      'RIGHT_OVERFLOW',
+    BOTH_SIDES_OVERFLOW: 'BOTH_SIDES_OVERFLOW'
+};
+
+var getCuttingType = function (originStr, diffIndex, maxStringLength) {
+    var maxStringOffsetFromMiddle = Math.floor(maxStringLength / 2);
+    var str                       = originStr.substr(1, originStr.length - 2);
+    var type                      = null;
+
+    //NOTE case: '123456'
+    //           '123945'
+    //               ^
+    if (str.length <= maxStringLength)
+        type = OVERFLOW_TYPES.NO_OVERFLOW;
+    //NOTE: case '1234...'
+    //           '1239...'
+    //               ^
+    else if (diffIndex < maxStringOffsetFromMiddle)
+        type = OVERFLOW_TYPES.RIGHT_OVERFLOW;
+    //NOTE: case '...456'
+    //           '...956;
+    //               ^
+    else if (str.length - diffIndex - 1 < maxStringOffsetFromMiddle)
+        type = OVERFLOW_TYPES.LEFT_OVERFLOW;
+    //NOTE: case '...345...'
+    //           '...395...'
+    //                ^
+    else
+        type = OVERFLOW_TYPES.BOTH_SIDES_OVERFLOW;
+
+    return type;
+};
+
+var getNewMarkerPosition = function (actualStr, diffIndex, maxStringLength) {
+    var maxLength                 = maxStringLength;
+    var maxStringOffsetFromMiddle = Math.floor(maxStringLength / 2);
+    var str                       = actualStr.substr(1, actualStr.length - 2);
+    var markerPosition            = 0;
+    var overflowType              = getCuttingType(actualStr, diffIndex, maxStringLength);
+
+    switch (overflowType) {
+        case OVERFLOW_TYPES.NO_OVERFLOW:
+            markerPosition = diffIndex;
+            break;
+        case OVERFLOW_TYPES.RIGHT_OVERFLOW:
+            markerPosition = diffIndex;
+            break;
+        case OVERFLOW_TYPES.LEFT_OVERFLOW:
+            markerPosition = diffIndex - (str.length - maxLength);
+            break;
+        case OVERFLOW_TYPES.BOTH_SIDES_OVERFLOW:
```

Comment:
`default:`? Actually it's why I don't like `switch` statements.
Maybe we can go with something like this instead of this function:
```js
const MARKER_POSITION_CALCULATORS = {
     [OVERFLOW_TYPES.noOverflow]: (actualStr, diffIndex) => diffIndex + 1,
     ...
     [OVERFLOW_TYPES.leftOverflow]: (actualStr, diffIndex) => {
          var str  = actualStr.substr(1, actualStr.length - 2); // What does it mean BTW????????
          return diffIndex - (str.length - maxLength);
     }
};
```